x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/metadata/TrainedModelMetadata.java

```diff
@@ -159,6 +172,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(FEATURE_IMPORTANCE_BASELINE.getPreferredName(), featureImportanceBaselines);
         }
         builder.field(HYPERPARAMETERS.getPreferredName(), hyperparameters);
+        builder.field(PLATFORM_ARCHITECTURE.getPreferredName(), platformArchitecture);// TODO version guard
```

Comment:
You don't need a version guard for JSON output, but don't print it if it's `null`. Printing `null` here is probably the cause of your test failure.

```suggestion
        if (platformArchitecture != null) {
            builder.field(PLATFORM_ARCHITECTURE.getPreferredName(), platformArchitecture);
        }
```